flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/TypesManager.java

```diff
@@ -0,0 +1,270 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.flowmodelgenerator.core;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import io.ballerina.compiler.api.ModuleID;
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.ArrayTypeSymbol;
+import io.ballerina.compiler.api.symbols.RecordFieldSymbol;
+import io.ballerina.compiler.api.symbols.RecordTypeSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.compiler.api.symbols.TypeDefinitionSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+import io.ballerina.flowmodelgenerator.core.model.Member;
+import io.ballerina.flowmodelgenerator.core.model.TypeData;
+import io.ballerina.projects.Package;
+import org.ballerinalang.model.types.TypeKind;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Generates text edits for the nodes that are requested to delete.
+ *
+ * @since 1.4.0
+ */
+public class TypesManager {
+    private static final Gson gson = new Gson();
+    private final Package currentPackage;
+
+    private static final Predicate<Symbol> supportedTypesPredicate = symbol -> {
+        if (symbol.getName().isEmpty()) {
+            return false;
+        }
+
+        if (symbol.kind().equals(SymbolKind.ENUM)) {
+            return true;
+        }
+
+        if (!symbol.kind().equals(SymbolKind.TYPE_DEFINITION)) {
+            return false;
+        }
+
+        switch (((TypeDefinitionSymbol) symbol).typeDescriptor().typeKind()) {
+            case RECORD, ARRAY, UNION, ERROR -> {
+                return true;
+            }
+            default -> {
+                return false;
+            }
+        }
+    };
+
+    public TypesManager(Package currentPackage) {
+        this.currentPackage = currentPackage;
+    }
+
+    public JsonElement getAllTypes() {
+        SemanticModel semanticModel = this.currentPackage.getDefaultModule().getCompilation().getSemanticModel();
```

Comment:
Why are we considering the default module instead of the module where the file exists?

Reply:
This is fixed via https://github.com/ballerina-platform/ballerina-dev-tools/pull/511/commits/c013f748fba0ec093bedb8669329221d9417208a